root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -4261,95 +4763,130 @@ const testData = [
              input_entity_type: 'artist',
     expected_relationship_type: 'viaf',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'http://viaf.org/viaf/152662182',
              input_entity_type: 'label',
     expected_relationship_type: 'viaf',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'http://viaf.org/viaf/185694157',
              input_entity_type: 'work',
     expected_relationship_type: 'viaf',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'http://viaf.org/viaf/16766997',
             expected_clean_url: 'http://viaf.org/viaf/16766997',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'http://viaf.org/viaf/16766997/',
             expected_clean_url: 'http://viaf.org/viaf/16766997',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'http://viaf.org/viaf/16766997/?test=true',
             expected_clean_url: 'http://viaf.org/viaf/16766997',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'http://viaf.org/viaf/16766997/#Rovics,_David',
             expected_clean_url: 'http://viaf.org/viaf/16766997',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'http://viaf.org/viaf/32197206/#Mozart,_Wolfgang_Amadeus,_1756-1791',
             expected_clean_url: 'http://viaf.org/viaf/32197206',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'https://www.viaf.org/viaf/16766997?test=1#Rovics,_David',
             expected_clean_url: 'http://viaf.org/viaf/16766997',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'viaf.org/viaf/16766997/',
             expected_clean_url: 'http://viaf.org/viaf/16766997',
+        expected_favicon_class: 'viaf-favicon',
   },
   {
                      input_url: 'www.viaf.org/viaf/16766997/',
             expected_clean_url: 'http://viaf.org/viaf/16766997',
+        expected_favicon_class: 'viaf-favicon',
   },
   // Videogam.in
   {
                      input_url: 'http://videogam.in/music/?id=PCCG-00486',
              input_entity_type: 'release',
     expected_relationship_type: 'otherdatabases',
+        expected_favicon_class: 'videogamin-favicon',
   },
   // Vimeo
   {
                      input_url: 'http://www.vimeo.com/1109226?pg=embed&sec=1109226',
              input_entity_type: 'recording',
     expected_relationship_type: 'streamingfree',
             expected_clean_url: 'https://vimeo.com/1109226',
+        expected_favicon_class: 'vimeo-favicon',
+  },
+  {
+                     input_url: 'https://vimeo.com/ondemand/inconcert/193518106?autoplay=1',
+            expected_clean_url: 'https://vimeo.com/ondemand/inconcert',
+        expected_favicon_class: 'vimeoondemand-favicon',
+  },
```

Comment:
Just a note for later on: This would really need additional tests when there is type restriction but no auto-select. For follow-up: https://tickets.metabrainz.org/browse/MBS-11980

Reply:
Agreed, thanks for adding the ticket.